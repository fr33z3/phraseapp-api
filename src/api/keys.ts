import {request} from 'client'
import {Key} from 'types/key'
import {CommonQueryOptions} from 'types/common'

type ListKeysOptions = CommonQueryOptions & {
  sort?: keyof Key
  order?: 'asc' | 'desc'
  q?: string
  locale_id?: string
}

export async function listKeys(projectId: string, options: ListKeysOptions = {}) {
  return request<Key[]>(`/projects/${projectId}/keys`, 'get', options)
}

export async function getKey(projectId: string, keyId: string, options: CommonQueryOptions = {}) {
  return request<Key>(`/projects/${projectId}/keys/${keyId}`, 'get', options)
}

type KeyParams = {
  branch: string
  name: string
  description: string
  plural: boolean
  name_plural: boolean
  data_type: boolean
  tags: string
  max_characters_allowed: number
  unformatted: boolean
  xml_space_preserve: boolean
  original_file: string
  localized_format_key: string
}

export async function createKey(projectId: string, params: KeyParams) {
  return request<Key>(`/projects/${projectId}/keys`, 'post', params)
}

export async function updateKey(projectId: string, keyId: string, params: KeyParams) {
  return request<Key>(`/projects/${projectId}/keys/${keyId}`, 'patch', params)
}
