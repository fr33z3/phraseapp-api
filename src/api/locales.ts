import {client} from 'client'
import {AccountLocale, Locale, LocaleStatistics} from 'types/locale'

export async function listAccountLocales(accountId: string) {
  const {data} = await client.get<AccountLocale>(`/accounts/${accountId}/locales`)
  return data
}

export async function listLocales(projectId: string) {
  const {data} = await client.get<Locale[]>(`/projects/${projectId}/locales`)
  return data
}

export async function getLocale(projectId: string, localeId: string) {
  const {data} = await client.get<Locale>(`/projects/${projectId}/locales/${localeId}`)
  return data
}

type LocaleParams = {
  branch?: string
  name: string
  code: string
  default?: boolean
  main?: boolean
  rtl?: boolean
  source_locale_id?: string
  unverify_new_translations?: boolean
  unverify_updated_translations?: boolean
  autotranslate?: boolean
}

type MutateLocaleResponse = Locale & {
  statistics: LocaleStatistics
}
export async function createLocale(projectId: string, params: LocaleParams) {
  const {data} = await client.post<MutateLocaleResponse>(`/projects/${projectId}/locales`, params)
  return data
}

export async function updateLocale(projectId: string, localeId: string, params: Partial<LocaleParams>) {
  const {data} = await client.patch<MutateLocaleResponse>(`/projects/${projectId}/locales/${localeId}`, params)
  return data
}

export async function deleteLocale(projectId: string, localeId: string) {
  const {data} = await client.delete(`/projects/${projectId}/locales/${localeId}`)
  return true
}
