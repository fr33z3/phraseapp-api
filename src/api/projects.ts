import {client} from 'client'
import {Project, ProjectParams} from 'types/project'

export async function listProjects() {
  const {data} = await client.get<Project[]>('/projects')
  return data
}

export async function getProject(id: string) {
  const {data} = await client.get<Project>(`/projects/${id}`)
  return data
}

export async function createProject(params: ProjectParams) {
  const {data} = await client.post<Project>('/projects', params)
  return data
}

export async function updateProject(id: string, params: Partial<ProjectParams>) {
  const {data} = await client.patch(`/projects/${id}`, params)
  return data
}

export async function deleteProject(id: string) {
  await client.delete(`/projects/${id}`)
  return true
}
