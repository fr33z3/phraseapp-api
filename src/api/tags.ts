import {client} from 'client'
import {Tag, TagParams, TagLocale, TagStatistics} from 'types/tag'

export async function listTags(projectId: string) {
  const {data} = await client.get<Tag[]>(`/projects/${projectId}/tags`)
  return data
}

export async function getTag(projectId: string, tagName: string) {
  const {data} = await client.get<Tag>(`/projects/${projectId}/tags/${tagName}`)
  return data
}

type CreateResponse = Tag & {
  statistics: {
    locale: TagLocale
    statistics: TagStatistics
  }
}

export async function createTag(projectId: string, params: TagParams) {
  const {data} = await client.post<CreateResponse>(`/projects/${projectId}/tags`, params)
  return data
}

export async function deleteTag(projectId: string, tagName: string) {
  await client.delete(`/projects/${projectId}/tags/${tagName}`)
  return true
}
