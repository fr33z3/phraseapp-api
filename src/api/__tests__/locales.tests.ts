import {createLocale, deleteLocale, getLocale, listAccountLocales, listLocales, updateLocale} from 'api/locales'

describe('Locale api', () => {
  describe('listAccountLocales', () => {
    it('returns all locales for a specified account', async () => {
      const result = await listAccountLocales('f996970154bcece65693ea658c160f0f')
      expect(result).toMatchSnapshot()
    })
  })

  describe('listLocales',() => {
    it('returns locales list', async () => {
      const result = await listLocales('77617b5199af590fde8642fb4178eaa1')
      expect(result).toMatchSnapshot()
    })
  })

  describe('createLocale', () => {
    it('creates new locale and returns locale data', async () => {
      const result = await createLocale('77617b5199af590fde8642fb4178eaa1', {name: 'Russian', code: 'ru'})
      expect(result).toMatchSnapshot()
    })
  })

  describe('updateLocale', () => {
    it('updates locale and returns locale data', async () => {
      const result = await updateLocale('77617b5199af590fde8642fb4178eaa1', '38dbf68906d0e5ab468eaef7d6d1a8ae', {name: 'RussianLanguage'})
      expect(result).toMatchSnapshot()
    })
  })

  describe('getLocale', () => {
    it('returns single locale data', async () => {
      const result = await getLocale('77617b5199af590fde8642fb4178eaa1', '0cd76a74f8850878c9e1ac09fa22925e')
      expect(result).toMatchSnapshot()
    })
  })

  describe('deleteLocale', () => {
    it('deletes locale', async () => {
      const result = await deleteLocale('77617b5199af590fde8642fb4178eaa1', '38dbf68906d0e5ab468eaef7d6d1a8ae')
      expect(result).toBeTruthy()
    })
  })
})
