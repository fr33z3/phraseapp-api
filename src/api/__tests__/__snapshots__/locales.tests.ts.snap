// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locale api createLocale creates new locale and returns locale data 1`] = `
Object {
  "code": "ru",
  "created_at": "2021-09-30T08:33:26Z",
  "default": false,
  "id": "38dbf68906d0e5ab468eaef7d6d1a8ae",
  "main": false,
  "name": "Russian",
  "plural_forms": Array [
    "zero",
    "one",
    "few",
    "many",
    "other",
  ],
  "rtl": false,
  "source_locale": null,
  "statistics": Object {
    "keys_total_count": 18,
    "keys_untranslated_count": 0,
    "missing_words_count": 23,
    "translations_completed_count": 0,
    "translations_unverified_count": 0,
    "unverified_words_count": 0,
    "words_total_count": 0,
  },
  "updated_at": "2021-09-30T08:33:26Z",
}
`;

exports[`Locale api getLocale returns single locale data 1`] = `
Object {
  "code": "en-US",
  "created_at": "2021-09-27T15:28:43Z",
  "default": true,
  "id": "0cd76a74f8850878c9e1ac09fa22925e",
  "main": false,
  "name": "en-US",
  "plural_forms": Array [
    "zero",
    "one",
    "other",
  ],
  "rtl": false,
  "source_locale": null,
  "statistics": Object {
    "keys_total_count": 18,
    "keys_untranslated_count": 0,
    "missing_words_count": null,
    "translations_completed_count": 18,
    "translations_unverified_count": 0,
    "unverified_words_count": 0,
    "words_total_count": 23,
  },
  "updated_at": "2021-09-27T15:28:43Z",
}
`;

exports[`Locale api listAccountLocales returns all locales for a specified account 1`] = `
Array [
  Object {
    "code": "ar-MA",
    "id": "2ef1309cdd0b10a35b3978340ed2a8b7",
    "name": "ar-MA",
    "project": Object {
      "created_at": "2021-09-27T15:28:39Z",
      "id": "77617b5199af590fde8642fb4178eaa1",
      "main_format": null,
      "name": "My Demo Application",
      "updated_at": "2021-09-30T08:56:25Z",
    },
  },
  Object {
    "code": "de-DE",
    "id": "4fafad4eb6557baf38d185cbd8511a39",
    "name": "de-DE",
    "project": Object {
      "created_at": "2021-09-27T15:28:39Z",
      "id": "77617b5199af590fde8642fb4178eaa1",
      "main_format": null,
      "name": "My Demo Application",
      "updated_at": "2021-09-30T08:56:25Z",
    },
  },
  Object {
    "code": "en",
    "id": "31d377e764d7d87928e45ade9a2f946a",
    "name": "English",
    "project": Object {
      "created_at": "2021-09-27T15:28:39Z",
      "id": "77617b5199af590fde8642fb4178eaa1",
      "main_format": null,
      "name": "My Demo Application",
      "updated_at": "2021-09-30T08:56:25Z",
    },
  },
  Object {
    "code": "en-US",
    "id": "0cd76a74f8850878c9e1ac09fa22925e",
    "name": "en-US",
    "project": Object {
      "created_at": "2021-09-27T15:28:39Z",
      "id": "77617b5199af590fde8642fb4178eaa1",
      "main_format": null,
      "name": "My Demo Application",
      "updated_at": "2021-09-30T08:56:25Z",
    },
  },
  Object {
    "code": "fr-FR",
    "id": "dc48ed5185edc91dda0e7189e9b84614",
    "name": "fr-FR",
    "project": Object {
      "created_at": "2021-09-27T15:28:39Z",
      "id": "77617b5199af590fde8642fb4178eaa1",
      "main_format": null,
      "name": "My Demo Application",
      "updated_at": "2021-09-30T08:56:25Z",
    },
  },
  Object {
    "code": "kpe-GN",
    "id": "6800a1d9184db966c2a90edce0a225e9",
    "name": "kpe-GN",
    "project": Object {
      "created_at": "2021-09-27T15:28:39Z",
      "id": "77617b5199af590fde8642fb4178eaa1",
      "main_format": null,
      "name": "My Demo Application",
      "updated_at": "2021-09-30T08:56:25Z",
    },
  },
  Object {
    "code": "th-TH",
    "id": "9dcea644b6c57c09c69d1f54a3b571f0",
    "name": "th-TH",
    "project": Object {
      "created_at": "2021-09-27T15:28:39Z",
      "id": "77617b5199af590fde8642fb4178eaa1",
      "main_format": null,
      "name": "My Demo Application",
      "updated_at": "2021-09-30T08:56:25Z",
    },
  },
  Object {
    "code": "ur-PK",
    "id": "f01a2a99707324af611d5edefccdff14",
    "name": "ur-PK",
    "project": Object {
      "created_at": "2021-09-27T15:28:39Z",
      "id": "77617b5199af590fde8642fb4178eaa1",
      "main_format": null,
      "name": "My Demo Application",
      "updated_at": "2021-09-30T08:56:25Z",
    },
  },
]
`;

exports[`Locale api listLocales returns locales list 1`] = `
Array [
  Object {
    "code": "ar-MA",
    "created_at": "2021-09-27T15:28:44Z",
    "default": false,
    "id": "2ef1309cdd0b10a35b3978340ed2a8b7",
    "main": false,
    "name": "ar-MA",
    "plural_forms": Array [
      "zero",
      "one",
      "two",
      "few",
      "many",
      "other",
    ],
    "rtl": true,
    "source_locale": null,
    "updated_at": "2021-09-27T15:28:44Z",
  },
  Object {
    "code": "de-DE",
    "created_at": "2021-09-27T15:28:43Z",
    "default": false,
    "id": "4fafad4eb6557baf38d185cbd8511a39",
    "main": false,
    "name": "de-DE",
    "plural_forms": Array [
      "zero",
      "one",
      "other",
    ],
    "rtl": false,
    "source_locale": null,
    "updated_at": "2021-09-27T15:28:43Z",
  },
  Object {
    "code": "en-US",
    "created_at": "2021-09-27T15:28:43Z",
    "default": true,
    "id": "0cd76a74f8850878c9e1ac09fa22925e",
    "main": false,
    "name": "en-US",
    "plural_forms": Array [
      "zero",
      "one",
      "other",
    ],
    "rtl": false,
    "source_locale": null,
    "updated_at": "2021-09-27T15:28:43Z",
  },
  Object {
    "code": "en",
    "created_at": "2021-09-30T08:01:55Z",
    "default": false,
    "id": "31d377e764d7d87928e45ade9a2f946a",
    "main": false,
    "name": "English",
    "plural_forms": Array [
      "zero",
      "one",
      "other",
    ],
    "rtl": false,
    "source_locale": null,
    "updated_at": "2021-09-30T08:01:55Z",
  },
  Object {
    "code": "fr-FR",
    "created_at": "2021-09-27T15:28:44Z",
    "default": false,
    "id": "dc48ed5185edc91dda0e7189e9b84614",
    "main": false,
    "name": "fr-FR",
    "plural_forms": Array [
      "zero",
      "one",
      "other",
    ],
    "rtl": false,
    "source_locale": null,
    "updated_at": "2021-09-27T15:28:44Z",
  },
  Object {
    "code": "kpe-GN",
    "created_at": "2021-09-27T15:28:44Z",
    "default": false,
    "id": "6800a1d9184db966c2a90edce0a225e9",
    "main": false,
    "name": "kpe-GN",
    "plural_forms": Array [
      "zero",
      "one",
      "two",
      "few",
      "many",
      "other",
    ],
    "rtl": false,
    "source_locale": null,
    "updated_at": "2021-09-27T15:28:44Z",
  },
  Object {
    "code": "ru",
    "created_at": "2021-09-30T08:03:59Z",
    "default": false,
    "id": "841099c7c4597555044959f464888d88",
    "main": false,
    "name": "Russian",
    "plural_forms": Array [
      "zero",
      "one",
      "few",
      "many",
      "other",
    ],
    "rtl": false,
    "source_locale": null,
    "updated_at": "2021-09-30T08:03:59Z",
  },
  Object {
    "code": "th-TH",
    "created_at": "2021-09-27T15:28:44Z",
    "default": false,
    "id": "9dcea644b6c57c09c69d1f54a3b571f0",
    "main": false,
    "name": "th-TH",
    "plural_forms": Array [
      "zero",
      "other",
    ],
    "rtl": false,
    "source_locale": null,
    "updated_at": "2021-09-27T15:28:44Z",
  },
  Object {
    "code": "ur-PK",
    "created_at": "2021-09-27T15:28:44Z",
    "default": false,
    "id": "f01a2a99707324af611d5edefccdff14",
    "main": false,
    "name": "ur-PK",
    "plural_forms": Array [
      "zero",
      "one",
      "other",
    ],
    "rtl": true,
    "source_locale": null,
    "updated_at": "2021-09-27T15:28:44Z",
  },
]
`;

exports[`Locale api updateLocale updates locale and returns locale data 1`] = `
Object {
  "code": "ru",
  "created_at": "2021-09-30T08:33:26Z",
  "default": false,
  "id": "38dbf68906d0e5ab468eaef7d6d1a8ae",
  "main": false,
  "name": "RussianLanguage",
  "plural_forms": Array [
    "zero",
    "one",
    "few",
    "many",
    "other",
  ],
  "rtl": false,
  "source_locale": null,
  "statistics": Object {
    "keys_total_count": 18,
    "keys_untranslated_count": 18,
    "missing_words_count": 23,
    "translations_completed_count": 0,
    "translations_unverified_count": 0,
    "unverified_words_count": 0,
    "words_total_count": 0,
  },
  "updated_at": "2021-09-30T08:42:47Z",
}
`;
